homeassistant/components/template/lock.py

```diff
@@ -142,7 +143,23 @@ def _update_state(self, result):
             return
 
         if isinstance(result, str):
-            self._state = result.lower()
+            if result.lower() in (
+                "true",
+                STATE_ON,
+                LockState.LOCKED,
```

Comment:
I'd not use `STATE_ON` or `LockState` to compare the rendered string. The latter is the rendered template, and not actually the entity state. That's why we need this translation before we can set the entity state.

Use a bare strings or a local constants.